pipelines/SpeciesTreeFromBusco/main.nf

```diff
@@ -328,110 +323,40 @@ process collateBusco {
 *@output path to aligned protein fasta
 *@output path to cDNA fasta
 */
-process alignProt {
-    label 'retry_with_8gb_mem_c1'
+process transAlign {
+    label 'retry_with_e64gb_mem_c16'
+
+    publishDir "${params.results_dir}/", pattern: "alignments/*_aln_*.fas", mode: "copy",  overwrite: true
 
     input:
-        val protFas
         path cdnas
     output:
         path "alignments/prot_aln_*.fas", emit: prot_aln
-        path cdnas, emit: cdnas
-   
-    script:
-    id = (protFas =~ /.*prot_(.*)\.fas$/)[0][1]
-    """
-    mkdir -p alignments
-    ${params.mafft_exe} --anysymbol --auto $protFas > alignments/prot_aln_${id}.fas
-    """
-}
-
-/**
-*@input path to protein alignment fasta
-*@output path to cDNA sequences fasta
-*@output path to codon alignment fasta
-*/
-process protAlnToCodon {
-    label 'rc_1Gb'
-
-    input:
-        path prot_aln
-        val cdna
-    output:
         path "alignments/codon_aln_*.fas", emit: codon_aln
-
-    script:
-    id = (prot_aln =~ /.*prot_aln_(.*)\.fas$/)[0][1]
-    """
-    mkdir -p alignments
-
-    # Filter out from the cDNA sequences which did not pass protein
-    # level filtering:
-    ${params.seqkit_exe} -j 5 fx2tab -n $prot_aln > prot.ids
-    ${params.seqkit_exe} grep -j 5 -n -f prot.ids $cdna > filtered_cdna.fas
-    # Convert AA to codon alignment:
-    (perl ${params.run_pal2nal_exe} ${params.pal2nal_exe} $prot_aln filtered_cdna.fas -output fasta) > alignments/codon_aln_${id}.fas
-    if [ -s alignments/codon_aln_${id}.fas ];
-    then
-        true;
-    else
-        echo "Codon alignment is empty!"
-    fi
-    """
-}
-
-/**
-*@input path to codon alignment fasta
-*@output path to codon alignment fasta with stop codons removed
-*/
-process removeStopCodons {
-    label 'rc_1Gb'
-
-    publishDir "${params.results_dir}/", pattern: "alignments/codon_aln_*.fas", mode: "copy",  overwrite: true
-
-    input:
-        path codon_aln
-    output:
-        path "alignments/codon_aln_*.fas", emit: codon_aln
-
+   
     script:
-    id = (codon_aln =~ /.*codon_aln_(.*)\.fas$/)[0][1]
+    id = (cdnas =~ /.*cdna_(.*)\.fas$/)[0][1]
     """
     mkdir -p alignments
-    if [ -s $codon_aln ];
-    then
-    java -jar ${params.macse_jar} -prog exportAlignment \
-    -align $codon_aln \
-    -codonForFinalStop --- \
-    -codonForInternalStop NNN \
-    -out_NT alignments/codon_aln_${id}.fas
+    set +e
+    ${params.pagan_exe} -s $cdnas --translate --threads 16 -o pagan_out
```

Comment:
How about not having to hard code the number of thread to pagan and get and pass it as a config parameter.

Reply:
In this case I belive hardcoding is the right thing to do. The reason is that the resource class is hardcoded and cannot be parametrised - it will request 16 cores, so specifying any other thread number does not make sense.